prov/gni/src/gnix_rma.c

```diff
@@ -845,19 +880,23 @@ static void __gnix_rma_more_fill_sub_htd(struct gnix_fab_req *req,
 {
 	int head_off, tail_len;
 	struct gnix_fid_ep *ep = req->gnix_ep;
+	int indirect = !!(req->flags & GNIX_RMA_INDIRECT);
 
 	head_off = req->rma.rem_addr & GNI_READ_ALIGN_MASK;
 	tail_len = (req->rma.rem_addr + req->rma.len)
 		& GNI_READ_ALIGN_MASK;
 
-	/* Get int_tx_buf if unaligned */
-	if ((req->int_tx_buf_e == NULL) && (head_off || tail_len)) {
+	/* Get int_tx_buf if unaligned and not indirect size. */
+	if ((req->int_tx_buf_e == NULL) &&
+			(head_off || tail_len) && !(indirect)) {
 		req->int_tx_buf_e = _gnix_ep_get_int_tx_buf(ep);
 		if (req->int_tx_buf_e == NULL) {
 			GNIX_FATAL(FI_LOG_EP_DATA, "RAN OUT OF INT_TX_BUFS");
 			/* TODO Create growable list of buffers */
 		}
 	} else {
+		GNIX_INFO(FI_LOG_EP_DATA,
+		"FI_MORE: Not filling htd since indirect or H/T Are aligned\n");
```

Comment:
Are you saying it should be misaligned instead of aligned?
I think i'll be deleting this statement anyways.